lib/removeUnusedElements.js

```diff
@@ -0,0 +1,235 @@
+            if (defined(indices) && indices > accessorId) {
+                primitive.indices--;
+            }
+        });
+    });
+
+    ForEach.skin(gltf, function(skin) {
+        if (defined(skin.inverseBindMatrices) && skin.inverseBindMatrices > accessorId) {
+            skin.inverseBindMatrices--;
+        }
+    });
+
+    ForEach.animation(gltf, function(animation) {
+        ForEach.animationSampler(animation, function(sampler) {
+            if (defined(sampler.input) && sampler.input > accessorId) {
+                sampler.input--;
+            }
+            if (defined(sampler.output) && sampler.output > accessorId) {
+                sampler.output--;
+            }
+        });
+    });
+};
+
+Remove.buffer = function(gltf, bufferId) {
+    var buffers = gltf.buffers;
+
+    buffers.splice(bufferId, 1);
+
+    ForEach.bufferView(gltf, function(bufferView) {
+        if (defined(bufferView.buffer) && bufferView.buffer > bufferId) {
+            bufferView.buffer--;
+        }
+    });
+};
+
+Remove.bufferView = function(gltf, bufferViewId) {
+    var bufferViews = gltf.bufferViews;
+
+    bufferViews.splice(bufferViewId, 1);
+
+    ForEach.accessor(gltf, function(accessor) {
+        if (defined(accessor.bufferView) && accessor.bufferView > bufferViewId) {
+            accessor.bufferView--;
+        }
+    });
+
+    ForEach.mesh(gltf, function(mesh) {
+        ForEach.meshPrimitive(mesh, function(primitive) {
+            if (defined(primitive.extensions) &&
+                defined(primitive.extensions.KHR_draco_mesh_compression)) {
+                if (primitive.extensions.KHR_draco_mesh_compression.bufferView > bufferViewId) {
+                    primitive.extensions.KHR_draco_mesh_compression.bufferView--;
+                }
+            }
+        });
+    });
+};
+
+/**
+ * Contains functions for getting a list of element ids in use by the glTF asset.
+ * @constructor
+ */
+function getListOfElementsIdsInUse() {}
+
+getListOfElementsIdsInUse.accessor = function(gltf) {
+    // Calculate accessor's that are currently in use.
+    var usedAccessorIds = {};
+
+    ForEach.mesh(gltf, function(mesh) {
+        ForEach.meshPrimitive(mesh, function(primitive) {
+            ForEach.meshPrimitiveAttribute(primitive, function(accessorId) {
+                usedAccessorIds[accessorId] = true;
+            });
+            ForEach.meshPrimitiveTarget(primitive, function(target) {
+                ForEach.meshPrimitiveTargetAttribute(target, function(accessorId) {
+                    usedAccessorIds[accessorId] = true;
+                });
+            });
+            var indices = primitive.indices;
+            if (defined(indices)) {
+                usedAccessorIds[indices] = true;
+            }
+        });
+    });
+
+    ForEach.skin(gltf, function(skin) {
+       if (defined(skin.inverseBindMatrices)) {
+           usedAccessorIds[skin.inverseBindMatrices] = true;
+       }
```

Comment:
The indentation in this block is slightly off.

Reply:
Done.